### PR TITLE
Gpio alternative

### PR DESCRIPTION
**Pull request by sigmashig**
The problem: some microcontrollers have a small number of GPIO pins and we should use some port expander. Using these port expanders requires replacement for GPIO methods. The virtual methods are added to PSX class for digitalRead/digitalWrite/pinMode. Declaration of these methods as virtual allows inheriting the PSX class and overwriting these methods by other ones. Nothing to do when you use standard GPIO pins.

### CODE REVIEW

File: platformio.ini
Comment (StarGate01) [inline platformio.ini:1]: I don't think this file should be part of the lib, as it is specific to your project I suppose?

**Comment by StarGate01** [comment]
Hey, thanks for the contribution! Good idea to add this functionality.

Just out of curiosity, what microcontroller are you using?

File: platformio.ini
Comment (sigmashig) [reply platformio.ini:1]: Oops. Sorry. Please, delete it.

**Comment by sigmashig** [comment]
> 

I'm using ESP32-WROOM DevKit. It has 15 pins available for users. Also, user have yet another 9 pins available with some restrictions, like 'Input only' or 'should be low level during boot'. I prefer don't use it when it is possible. I'm using this microcontroller for RC car control. This controller has LoRa module connected. (7 pins), extra UART (2 pins), I2C display + battery control (2 pins), So, after that, I have 4 pins free and it is enough for the gamepad. But I would like to connect the keyboard and some LEDs to indicate status. I'm going to do it with I2C port expander, but this expander requires an extra pin for the interrupt line. 
BTW, I moved the gamepad to the port expander. It is working well, except DAT line. I have no idea, why it doesn't work on expander. The digitalRead returns 0 in any case. Maybe, it is my bug or the speed of the port expander is not enough. I will test it later. In any case, I successfully moved the rest 3 lines to the port expander.

**Comment by StarGate01** [comment]
Thanks for cleaning up the PR. Looks good to me, cheers!

**Pull request merged**